analyze-network-with-subnetwork-trace/src/main/java/com/esri/arcgismaps/sample/analyzenetworkwithsubnetworktrace/MainActivity.kt

```diff
@@ -0,0 +1,512 @@
+            operatorsList?.get(operatorPosition)?.let { comparisonOperator ->
+                // NOTE: You may also create a UtilityNetworkAttributeComparison
+                // with another NetworkAttribute
+                var expression: UtilityTraceConditionalExpression =
+                    UtilityNetworkAttributeComparison(
+                        sourceAttribute,
+                        comparisonOperator,
+                        otherValue
+                    )
+                (traceConfiguration.traversability?.barriers as? UtilityTraceConditionalExpression)?.let { otherExpression ->
+                    // NOTE: You may also combine expressions with UtilityTraceAndCondition
+                    expression = UtilityTraceOrCondition(otherExpression, expression)
+                }
+                traceConfiguration.traversability?.barriers = expression
+                expressionTextView.text = expressionToString(expression)
+            }
+        }
+    }
+
+    /**
+     * Show the UI of the given [id] and hide the others which share the same space.
+     */
+    private fun setVisible(id: Int) {
+        when (id) {
+            valuesBackgroundView.id -> {
+                valuesBackgroundView.visibility = View.VISIBLE
+                valueBooleanButton.visibility = View.GONE
+                valuesEditText.visibility = View.GONE
+            }
+            valuesEditText.id -> {
+                valuesEditText.visibility = View.VISIBLE
+                valueBooleanButton.visibility = View.GONE
+                valuesBackgroundView.visibility = View.GONE
+            }
+            valueBooleanButton.id -> {
+                valueBooleanButton.visibility = View.VISIBLE
+                valuesBackgroundView.visibility = View.GONE
+                valuesEditText.visibility = View.GONE
+            }
+        }
+    }
+
+    /**
+     * Run the network trace with the parameters and display the result in an alert dialog
+     * when the [traceButton] is clicked.
+     */
+    fun trace(traceButton: View) {
+        if (utilityNetwork.loadStatus.value != LoadStatus.Loaded) {
+            return showError("Utility network is not loaded")
+        }
+
+        // set the utility trace parameters
+        val parameters = UtilityTraceParameters(
+            UtilityTraceType.Subnetwork,
+            listOf(startingLocation).requireNoNulls()
+        ).apply {
+            // set the utility trace configuration options to include
+            traceConfiguration = utilityTraceConfiguration?.apply {
+                includeBarriers = barriersCheckbox.isChecked
+                includeContainers = containersCheckbox.isChecked
+            }
+        }
+
+        // launch trace in a coroutine scope
+        lifecycleScope.launch {
+            showLoadingDialog(true)
+            val utilityTraceResults = utilityNetwork.trace(parameters).getOrElse {
+                return@launch showError(
```

Comment:
before this, the dialog needs to dismiss

Reply:
In theory, `traceButton` will never be able to be clicked if there is a dialog being shown. So, launching a dialog here should be safe.